src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -566,7 +566,7 @@ SyntaxNode GetMatchingRoot(SyntaxNode body)
                 case SyntaxKind.ThisConstructorInitializer:
                 case SyntaxKind.BaseConstructorInitializer:
                 case SyntaxKind.ConstructorDeclaration:
-                    var newConstructor = (ConstructorDeclarationSyntax)newBody.Parent;
+                    var newConstructor = (ConstructorDeclarationSyntax)(newBody.Parent.IsKind(SyntaxKind.ArrowExpressionClause) ? newBody.Parent.Parent : newBody.Parent);
```

Comment:
>newBody.Parent [](start = 72, length = 14)

do we need a null check for newBody.Parent?

Reply:
Why? These nodes (ctor initilizer, ctor declaration) always have parent.